packages/apollos-ui-notifications/src/Provider.js

```diff
@@ -56,14 +58,26 @@ class NotificationsInit extends Component {
     );
   }
 
-  componentDidMount() {
-    OneSignal.init(this.props.oneSignalKey, {
-      kOSSettingsKeyAutoPrompt: false,
-    });
-    OneSignal.addEventListener('received', this.onReceived);
-    OneSignal.addEventListener('opened', this.onOpened);
-    OneSignal.addEventListener('ids', this.onIds);
-    OneSignal.setSubscription(true);
+  async componentDidMount() {
+    // One Signal 4.x
+    if (OneSignal.setAppId) {
```

Comment:
```suggestion
    if (OneSignal.setAppId && this.props.oneSignalKey) {
```

to avoid this and to make the prop optional

<img width="473" alt="Screen Shot 2021-06-10 at 3 59 49 PM" src="https://user-images.githubusercontent.com/2659478/121589567-36532d00-ca05-11eb-8ecf-8a86eba8bfec.png">

Reply:
actually it's gonna need to go one level up so when there's no prop, v3 code doesn't try to run

Reply:
v3 doesn't error, so I think your change is good

Reply:
well, if you're on v4 and don't pass the prop, it's gonna drop to v3 code and try to run `Onesignal.init` which will error because it doesn't exist.

Reply:
nevermind, it just comes through as an unresolved promise.

<img width="473" alt="Screen Shot 2021-06-10 at 4 58 51 PM" src="https://user-images.githubusercontent.com/2659478/121596272-39eab200-ca0d-11eb-9e36-35be0786fc2d.png">